api/modules/budgeting.js

```diff
@@ -0,0 +1,103 @@
+const db = require('../models');
+const moment = require('moment')
+
+const budgeting = module.exports = (() => {
+
+    const createPayment = async ({ paymentInformation }) => {
+        const client = await getClientWithExternalId({ id: paymentInformation.customer_id })
+        if (client) {
+            return db.models.Payment.create({
+                amount: paymentInformation.amount,
+                external_uuid: paymentInformation.external_uuid,
+                date_incurred: moment(paymentInformation.date_incurred['_d']).format('YYYY-MM-DD HH:mm:ss'),
```

Comment:
What is the `_d` notation here? It seems confusing and not obvious what it does or where it comes from.

If it's possible to remove it away that'd be best but if not a comment should be left so the reader is not confused why it is there

Reply:
@sofiarm21 Can you help us here? so we can find the best solution!

Reply:
@mecoccaro yeah so `['_d']` notation is used to access the instance of the Date object that backs the moment object. So in this case, `date_incurred` is a moment object and we are accessing its internal property `_d`. We shouldn't be accessing internals like this and instead, we should be using the functions that the moment library provides to us.  
So instead of `moment(paymentInformation.date_incurred['_d']).format('YYYY-MM-DD HH:mm:ss')` this could be refactored to something like this `paymentInformation.date_incurred.format('YYYY-MM-DD HH:mm:ss')` 
Ik there are some places around the app that we are accessing the internals of the moment object but this should be refactored at some point. 

[Here](https://momentjs.com/guides/) is the documentation about the moment library in case you haven't checked it out already.